src/core/server.js

```diff
@@ -676,6 +676,17 @@ Server.prototype._handleResponse = function(message) {
     const result = message.result;
     const responseEvent = 'response_' + command;
 
+    if (command === 'submit' && result.engine_result === 'tefPAST_SEQ' &&
```

Comment:
Yes, that's the problem here. So I think that @wltsmrz is right, we need to drop re-request logic and make timeout for requests.